### ISSUE TITLE
[Bug]: Second and subsequent calls to plt.polar() return an Axis object and do not plot

### ISSUE THREAD

**Issue by beckja** [matplotlib/matplotlib#21113]
### Bug summary

When plt.polar() is called while the current axes are a PolarAxes object, the call just returns the axes again without plotting the new data.

The code for polar() in pyplot.py is easily fixed by changing "return ax" to "return ax.plot(*args, **kwargs)".

### Code for reproduction

```python
import matplotlib.pyplot as plt

h1 = plt.polar([0],[1],'ro')

h2 = plt.polar([1.57],[.5], 'bo')

plt.show()
```

### Actual outcome

Only the first point is plotted.  Note that h1 is a list of length 1 of Line2D objects, while h2 is a PolarAxesSubplot object.

### Expected outcome

Both points should be plotted.  Both h1 and h2 should be lists of length 1 of Line2D objects.

### Operating system

Windows

### Matplotlib Version

3.4.3

### Python version

3.9

### Installation

conda

**Comment by beckja** [comment]
Suggested slightly cleaner version:

def polar(*args, **kwargs):
    """
    Make a polar plot.

    call signature::

      polar(theta, r, **kwargs)

    Multiple *theta*, *r* arguments are supported, with format strings, as in
    `plot`.
    """
    # If an axis already exists, check if it has a polar projection
    if gcf().get_axes():
        ax = gca()
        if not isinstance(ax, PolarAxes):
            _api.warn_external('Trying to create polar plot on an Axes '
                               'that does not have a polar projection.')
    else:
        ax = axes(projection="polar")
    return ax.plot(*args, **kwargs)

**Comment by timhoffm** [comment]
Yes. Do you want to make a pull request?

**Comment by timhoffm** [comment]
Note: This is a regression from #19153 introduced into 3.4.0.

**Comment by beckja** [comment]
I'm working the pull request and will submit shortly.

**Comment by tacaswell** [comment]
Thank you @beckja !

**Comment by QuLogic** [comment]
Closed by #21131.
